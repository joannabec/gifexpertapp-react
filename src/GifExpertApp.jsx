import React, { useState } from 'react';
import AddCategories from './components/AddCategories';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  return (
    <>
      <h2>GifExpertApp</h2>
      {/* Componente del input */}
      <AddCategories setCategories={setCategories} />
      <hr />

      <ol>
        {/* Componente donde van el tÃ­tulo y las cards */}
        {categories.map((item) => (
          <GifGrid key={item} category={item} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
