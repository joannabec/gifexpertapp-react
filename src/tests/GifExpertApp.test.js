/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Testeando GifExpertApp', () => {
  test('Debería renderizar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería mostrar una lista de categorías', () => {
    const categories = ['Avatar', 'Hola'];
    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
