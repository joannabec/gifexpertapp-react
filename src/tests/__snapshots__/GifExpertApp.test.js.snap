// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando GifExpertApp Debería mostrar una lista de categorías 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Avatar"
      key="Avatar"
    />
    <GifGrid
      category="Hola"
      key="Hola"
    />
  </ol>
</Fragment>
`;

exports[`Testeando GifExpertApp Debería renderizar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
